execution/executor/src/block_executor.rs

```diff
@@ -220,19 +220,35 @@ where
         sync_commit: bool,
     ) -> Result<(), Error> {
         let _timer = APTOS_EXECUTOR_COMMIT_BLOCKS_SECONDS.start_timer();
+
+        // Ensure the block ids are not empty
+        if block_ids.is_empty() {
+            return Err(anyhow::anyhow!("Cannot commit 0 blocks!").into());
+        }
+
+        // Check for any potential retries
         let committed_block = self.block_tree.root_block();
         if committed_block.num_persisted_transactions()
             == ledger_info_with_sigs.ledger_info().version() + 1
         {
-            // a retry
             return Ok(());
         }
 
+        // Ensure the last block id matches the ledger info block id to commit
         let block_id_to_commit = ledger_info_with_sigs.ledger_info().consensus_block_id();
         info!(
             LogSchema::new(LogEntry::BlockExecutor).block_id(block_id_to_commit),
             "commit_block"
         );
+        let last_block_id = *block_ids.last().unwrap();
+        if last_block_id != block_id_to_commit {
+            return Err(anyhow::anyhow!(
+                "Block id to commit ({:?}) does not match last block id ({:?})!",
+                block_id_to_commit,
+                last_block_id
+            )
+            .into());
+        }
```

Comment:
I'm not exactly sure how it works with reconfig suffixes @zekun000 to decide.

Reply:
this block id still matches, those checks probably should be assertion than error, it's just invariant

Reply:
K. Then I agree, I'd rather panic here.

Reply:
Cool, I've updated it to panic! 😄